src/Stratis.Bitcoin.Features.Consensus/RuleContextExtention.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using NBitcoin;
+using Stratis.Bitcoin.Consensus;
+using Stratis.Bitcoin.Consensus.Rules;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.Features.Consensus
+{
+    public class PosRuleContext
+    {
+        public BlockStake BlockStake { get; set; }
+
+        public Money TotalCoinStakeValueIn { get; set; }
+
+        public uint256 HashProofOfStake { get; set; }
+
+        public uint256 TargetProofOfStake { get; set; }
+    }
+
+    public static class RuleContextExtention
+    {
+        public static void SetStake(this RuleContext ruleContext)
+        {
+            ruleContext.SetItem(new PosRuleContext
+            {
+                BlockStake = new BlockStake(ruleContext.Item<BlockValidationContext>().Block)
```

Comment:
Everywhere else you have context.BlockValidationContext, not Item

Reply:
I will fix it back